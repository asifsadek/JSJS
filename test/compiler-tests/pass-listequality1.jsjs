val xs = [1, 2, 3];
val ys = [1, 2, 3];

// same
val _ = if xs == ys then print("same lists") else print("diff lists");
val _ = if ((10 :: []) == (10 :: [])) then print("same lists") else print("diff lists");


// diff
val _ =
  if (20 :: []) == (10 :: [])
  then print("same lists")
  else print("diff lists");

val _ =
  if [1, 2, 3] == [1, 4, 5]
  then print("same lists")
  else print("diff lists");
