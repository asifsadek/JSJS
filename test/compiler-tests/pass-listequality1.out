same lists
same lists
diff lists
diff lists
